src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -857,8 +857,13 @@ def getTrustSitelists(self):
 
         Get the input and pileup flag for 'trust site lists' in the task.
         """
-        return {'trustlists': self.data.constraints.sites.trustlists,
-                'trustPUlists': self.data.constraints.sites.trustPUlists}
+        
+        # handle backward compatibility for the request which doesn't contain trustPUlists
+        if hasattr(self.data.constraints.sites, "trustPUlists"):
```

Comment:
In addition to Alan's point that you switched from dict to tuple, why don't you wrap the original code in try/except AttributeError: and then
```
return {'trustlists': self.data.constraints.sites.trustlists,
            'trustPUlists': False}
```